Add 1.35 release notes (#1063)
okteto/docs: src/content/release-notes.mdx

id: release-notes
---

## 1.35.0

1 August 2025

This version is compatible with Kubernetes versions 1.31 to 1.33
Okteto Chart release 1.35 is designed to work with [Okteto CLI 3.10.x](https://github.com/okteto/okteto/releases/tag/3.10.0)

### New Features {#new-features-1.35}

> - CIDR-based traffic filtering is now supported for BYOC (Bring Your Own Cluster) environments. Use this to improve security by restricting access to Okteto services to specific IP ranges.

Reviewer: Is this something we want to mention in the release notes taking into account we didn't do changes in the chart more than the point about supporting `loadBalancerSourceRanges` in BuildKit?

Customers in BYOC model cannot do the changes on their own. Maybe this is something to add into some BYOC specific instead of here?

If we add it in the docs, I think it is important to highlight the restrictions (I don't know the details, but I do know there are some)

> - Added support for [Kubernetes 1.33](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.33.md) (support for 1.30 has been removed) and [Amazon Linux 2023](https://github.com/amazonlinux/amazon-linux-2023)

Reviewer: I would set this the first element in the list of features. Also, would it make sense to do a callout for AWS users to the content you are adding [here](https://github.com/okteto/docs/pull/1061)?

Author: Yes, good call. Let me get that PR cleaned up and I will rebase and link to it

> - Namespace deletion now includes a configurable timeout to help with stuck cleanups  

Reviewer: Namespace deletion always had a timeout flag to specify, but it wasn't propagated to dev environments with graceful destroy (the ones with a `destroy` section in the manifest). So, if the dev environment destruction was taking more than 5 minutes, the destroy namespace operation was failing. I think it might be relevant to give those details

> -	Added support for loadBalancerSourceRanges in the BuildKit service configuration to better control external access  

Reviewer: ```suggestion
- Added support for loadBalancerSourceRanges in the BuildKit service configuration to better control external access  
```

### Bug Fixes {#bug-fixes-1.35}

> - Fixed an issue that prevented some development environments from waking up as expected  

Reviewer: ```suggestion
- Fixed an issue that prevented some development environments from waking up as expected  when there were dependency cycles between dev environments
```

To provide a bit more of context to users

> - Improved AWS IAM Role regex handling for tighter validation

Reviewer: ```suggestion
- Improved AWS IAM Role regex handling for tighter validation on Private Registry Credentials and Cloud Credentials
```

### New Features {#new-features-1.35}
- Added support for [Kubernetes 1.33](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.33.md) (support for 1.30 has been removed) and [Amazon Linux 2023](https://github.com/amazonlinux/amazon-linux-2023). [Please follow our upgrade guide](self-hosted/manage/upgrade.mdx#upgrading-to-okteto-135x--kubernetes-133-support-and-amazon-linux-2023-al2023compatibility) when moving to Amazon Linux 2023
- CIDR-based traffic filtering is now supported for BYOC (Bring Your Own Cluster) environments. Use this to improve security by restricting access to Okteto services to specific IP ranges.
- Namespace deletion now properly applies the timeout value to dev environments that use a `destroy` section in their manifest. Previously, if a dev environment took longer than 5 minutes to destroy gracefully, the overall namespace deletion would fail, even if a longer timeout was specified. <!-- 9530 -->
- Added support for `loadBalancerSourceRanges` in the BuildKit service configuration to better control external access

> - [Okteto CLI 3.10.0](https://github.com/okteto/okteto/releases/tag/3.10.0): Okteto now supports inheriting Kubernetes `nodeSelector` and resource settings in Development Environments. When omitted from the `okteto.yaml` manifest, these values can be pulled from the base Kubernetes resources using the `OKTETO_INHERIT_KUBERNETES_RESOURCES` and `OKTETO_INHERIT_KUBERNETES_NODESELECTOR` environment variables.

Reviewer: ```suggestion
- [Okteto CLI 3.10.0](https://github.com/okteto/okteto/releases/tag/3.10.0): Okteto now supports inheriting Kubernetes `nodeSelector` and resource settings in Development Environments. When omitted from the `okteto.yaml` manifest, these values can be pulled from the base Kubernetes resources using the `OKTETO_INHERIT_KUBERNETES_RESOURCES` and `OKTETO_INHERIT_KUBERNETES_NODESELECTOR` feature flags.
```

We usually call them feature flags in other release notes (and we usually add a link to the feature flag section)

### Improvements {#improvements-1.35}

> - [Okteto CLI 3.10.0](https://github.com/okteto/okteto/releases/tag/3.10.0): Improved the `okteto deploy` command to avoid rebuilding all images when only a subset of services is being deployed.

Reviewer: ```suggestion
- [Okteto CLI 3.10.0](https://github.com/okteto/okteto/releases/tag/3.10.0): Improved the `okteto deploy` command to avoid rebuilding all images when deploying a compose and only a subset of services is being deployed.
```

Would it make sense to add that to avoid misunderstandings?
